scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -311,11 +315,13 @@ class FossId(
                     }
                 }
 
-                checkScan(scanCode)
-                val rawResults = getRawResults(scanCode)
-                val resultsSummary = createResultSummary(startTime, rawResults)
+                if (!noWait) {
+                    checkScan(scanCode)
+                    val rawResults = getRawResults(scanCode)
+                    val resultsSummary = createResultSummary(startTime, rawResults)
 
-                results.getOrPut(pkg) { mutableListOf() } += resultsSummary
+                    results.getOrPut(pkg) { mutableListOf() } += resultsSummary
```

Comment:
IIUC, there is currently no indication that a package has been processed if `noWait` is *true*. Would it make sense to have one?

It is clear that no actual result is available yet, but could there be a kind of dummy result - ideally with information from where the final result can be obtained? Does the ORT data model offer anything that could be used for this purpose? (The support for asynchronous scanners is rather limited, I assume.)

Reply:
I thought about creating an ORT issue with a level of `HINT` with something like:
`Scanner has run in asynchronous mode. Scan results are available on the FossID instance.`

but since issue is a sub property from `ScanSummary`, it would have meant that this issue repeats identically for each package.

Now if I cross it with your first comment, we could change the issue content to:

`Package xyz has been scanned in asynchronous mode. Scan results are available on the FossID instance.`

What to you think about it ?

Reply:
This would be fine with me as a work-around - until asynchronous scanners are treated as first-class citizens in the ORT model.